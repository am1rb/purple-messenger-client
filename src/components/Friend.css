
.candidate, .friend
{
	padding: 10px 10px;
	display: flex;
}

.friend:hover
{
	background: #e9e9e9;
}

.friendImage
{
	width: 50px;
	height: 50px;
	border-radius: 50%;
	/*margin-right: 10px;*/
}

.friendInfo
{
	width: 100%;
	margin: 0 1em;
}

.candidate
{
	cursor: pointer;
}

.candidateInfo
{
	/*display: flex;*/
	margin: 0 1em;
	/*position: relative;*/
	width: 100%;
}

.addIcon
{
	margin: 0 0.5em;
}

.disabled
{
	cursor: default;
}