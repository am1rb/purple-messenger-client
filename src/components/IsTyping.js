// @flow

import React from 'react'

import styles from './IsTyping.css'

type Props = {};


export default class IsTyping extends React.PureComponent<Props>
{
	render()
	{
		return (
			<span className={styles.label}>
				is typing...
			</span>
		)
	}
}