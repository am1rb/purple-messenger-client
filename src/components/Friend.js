// @flow

import React from 'react'

import {Link} from 'react-router-dom'
import classNames from 'classnames'

import IsTyping from './IsTyping'

import styles from './Friend.css'
import defaultImage from '../assets/default-user.png'


type Props = {
	id: string,
	fullName: string,
	isTyping: boolean,
	lastMessage: string,
};

export default class Friend extends React.PureComponent<Props>
{
	render()
	{
		return (
			<Link to={"/friend/"+this.props.id+"/"} className={styles.friend}>
				<img src={defaultImage} className={styles.friendImage} alt={this.props.fullName} />
				<div className={styles.friendInfo}>
					<div className={styles.friendName}>
						{this.props.fullName}
					</div>
					<div className={styles.message}>
						{this.props.isTyping ? <IsTyping /> : this.props.lastMessage}
					</div>
				</div>
			</Link>
		)
	}
}