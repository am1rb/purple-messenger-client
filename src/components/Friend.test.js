import renderer from 'react-test-renderer'
import React from 'react'
import { MemoryRouter } from 'react-router'

import IsTyping from './IsTyping'
import Friend from './Friend'



describe("<Friend /> component tests", () => {

	it("Should has typing mode", () => {
		
		const {root} = renderer.create(
			<MemoryRouter>
				<Friend
					isTyping={true}
				/>
			</MemoryRouter>
		)

		expect(root.findAllByType(IsTyping).length).toBe(1)
	})

	it("Should not regress", () => {
		
		const friend = renderer.create(
			<MemoryRouter>
				<Friend
					id="testId"
					fullName="Amir"
					lastMessage="This is last message"
					isTyping={false}
				/>
			</MemoryRouter>
		)

		expect(friend.toJSON()).toMatchSnapshot();
	})

})