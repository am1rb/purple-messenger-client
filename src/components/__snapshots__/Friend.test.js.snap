// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Friend /> component tests Should has valid snapshot 1`] = `
<a
  className={undefined}
  href="/friend/testId/"
  onClick={[Function]}
>
  <img
    alt="Amir"
    className={undefined}
    src="default-user.png"
  />
  <div
    className={undefined}
  >
    <div
      className={undefined}
    >
      Amir
    </div>
    <div
      className={undefined}
    >
      This is last message
    </div>
  </div>
</a>
`;

exports[`<Friend /> component tests Should not regress 1`] = `
<a
  className={undefined}
  href="/friend/testId/"
  onClick={[Function]}
>
  <img
    alt="Amir"
    className={undefined}
    src="default-user.png"
  />
  <div
    className={undefined}
  >
    <div
      className={undefined}
    >
      Amir
    </div>
    <div
      className={undefined}
    >
      This is last message
    </div>
  </div>
</a>
`;
